pkg/diffs/diffs.go

```diff
@@ -161,33 +161,58 @@ func GetImagesPostsubmitsForCiopConfigs(prowConfig *prowconfig.Config, ciopConfi
 	return ret
 }
 
-func GetPresubmitsForCiopConfigs(prowConfig *prowconfig.Config, ciopConfigs config.DataByFilename, affectedJobs map[string]sets.String, logger *logrus.Entry) config.Presubmits {
-	ret := config.Presubmits{}
+func GetJobsForCiopConfigs(prowConfig *prowconfig.Config, ciopConfigs config.DataByFilename, affectedJobs map[string]sets.String, logger *logrus.Entry) (config.Presubmits, config.Periodics) {
+	var presubmits config.Presubmits
+	var periodics config.Periodics
+
+	skip := func(job prowconfig.JobBase, prefix string, cfgFile string) bool {
+		if job.Agent != string(pjapi.KubernetesAgent) {
+			return true
+		}
+		if !strings.HasPrefix(job.Name, prefix) {
+			return true
+		}
+		testName := strings.TrimPrefix(job.Name, prefix)
+
+		affectedJob, ok := affectedJobs[cfgFile]
+		if ok && !affectedJob.Has(testName) {
+			return true
+		}
+		return false
+
+	}
 
 	for _, data := range ciopConfigs {
 		orgRepo := fmt.Sprintf("%s/%s", data.Info.Org, data.Info.Repo)
-		jobNamePrefix := data.Info.JobName(jobconfig.PresubmitPrefix, "")
+		cfgFile := data.Info.Basename()
 		for _, job := range prowConfig.JobConfig.PresubmitsStatic[orgRepo] {
-			if job.Agent != string(pjapi.KubernetesAgent) {
+			if skip(job.JobBase, data.Info.JobName(jobconfig.PresubmitPrefix, ""), cfgFile) {
 				continue
 			}
-			if !strings.HasPrefix(job.Name, jobNamePrefix) {
-				continue
-			}
-			testName := strings.TrimPrefix(job.Name, jobNamePrefix)
 
-			affectedJob, ok := affectedJobs[data.Info.Basename()]
-			if ok && !affectedJob.Has(testName) {
+			selectionFields := logrus.Fields{LogRepo: orgRepo, LogJobName: job.Name, LogReasons: "ci-operator config changed"}
+			logger.WithFields(selectionFields).Info(ChosenJob)
+			if presubmits == nil {
```

Comment:
Why bother initializing them to `nil` if you're just going to do this?

Reply:
We may skip all jobs and then we return nil. But I guess it's simpler to initialize to an empty map and return it instead of nil, easier code :+1:

Reply:
I don't think that I would personally expect to personally distinguish between a map that's `nil` versus one that has no elements ...

Reply:
It's just annoying in tests where you cannot leave the expected "empty" item initialized ;)